cms/djangoapps/export_course_metadata/signals.py

```diff
@@ -0,0 +1,40 @@
+"""
+This file exports metadata about the course.
+Initially the export goes to S3,
+but the storage class could be used for other backends in the future.
+
+This data is initially being used by Braze Connected Content to include
+section highlights in emails, but may be used for other things in the future.
+"""
+
+import json
+import logging
+
+from django.core.files.base import ContentFile
+from django.dispatch import receiver
+from openedx.core.djangoapps.schedules.content_highlights import get_all_course_highlights
+from xmodule.modulestore.django import SignalHandler
+
+from .storage import course_metadata_export_storage
+from .toggles import EXPORT_COURSE_METADATA_FLAG
+
+log = logging.getLogger(__name__)
+
+
+@receiver(SignalHandler.course_published)
+def export_course_metadata(sender, course_key, **kwargs):  # pylint: disable=unused-argument
+    """
+    Export course metadata on course publish.
+
+    File format
+    '{"highlights": [["week1highlight1", "week1highlight2"], ["week1highlight1", "week1highlight2"], [], []]}'
+    To retrieve highlights for week1, you would need to do
+    course_metadata['highlights'][0]
+
+    This data is initially being used by Braze Connected Content to include
+    section highlights in emails, but may be used for other things in the future.
+    """
+    if EXPORT_COURSE_METADATA_FLAG.is_enabled():
+        highlights = get_all_course_highlights(course_key)
+        highlights_content = ContentFile(json.dumps({"highlights": highlights}))
+        course_metadata_export_storage.save('course_metadata_export/{}.json'.format(course_key), highlights_content)
```

Comment:
Do we know how this will actually perform? Maybe it's dependent on the storage we are using? But is it going to append, overwrite, etc.? Might be worth mentioning that either in this function definition (if all storages should perform the same way) or in the storage class definition if it's specific to each storage backend in which case we should document how S3 behaves.

Reply:
pretty sure it is going to overwrite though per this comment https://github.com/edx/edx-platform/pull/26744/files#r585544848 i would only actually test the behavior with the waffle flag once this is deployed. 
why would we document the behavior of the django-storages save method though?  
it seems like that's a detail of the library (and potentially subject to change if we upgrade?)